Assets/MRTK/Services/InputSystem/FocusProvider.cs

```diff
@@ -1226,40 +1226,48 @@ private static void QueryScene(IMixedRealityPointer pointer, IMixedRealityRaycas
-                                GameObject closest = null;
-                                float closestDistance = Mathf.Infinity;
-                                Vector3 objectHitPoint = testPoint;
-
-                                for (int colliderIndex = 0; colliderIndex < numColliders; colliderIndex++)
+                                int numColliders = UnityEngine.Physics.OverlapSphereNonAlloc(pointer.Rays[i].Origin, pointer.SphereCastRadius, colliders, layerMask);
+                                totalNumColliders += numColliders;
+                                if (numColliders > 0)
                                 {
-                                    Collider collider = colliders[colliderIndex];
-                                    // Policy: in order for an collider to be near interactable it must have
-                                    // a NearInteractionGrabbable component on it.
-                                    // FIXME: This is assuming only the grab pointer is using SceneQueryType.SphereOverlap,
-                                    //        but there may be other pointers using the same query type which have different semantics.
-                                    if (collider.GetComponent<NearInteractionGrabbable>() == null)
+                                    if (totalNumColliders >= maxQuerySceneResults)
                                     {
-                                        continue;
+                                        Debug.LogWarning($"Maximum number of {totalNumColliders} colliders found in FocusProvider overlap query. Consider increasing the focus query buffer size in the input profile.");
                                     }
-                                    // From https://docs.unity3d.com/ScriptReference/Collider.ClosestPoint.html
-                                    // If location is in the collider the closestPoint will be inside.
-                                    Vector3 closestPointToCollider = collider.ClosestPoint(testPoint);
-                                    float distance = (testPoint - closestPointToCollider).sqrMagnitude;
-                                    if (distance < closestDistance)
+                                    for (int colliderIndex = 0; colliderIndex < numColliders; colliderIndex++)
                                     {
-                                        closestDistance = distance;
-                                        closest = collider.gameObject;
-                                        objectHitPoint = closestPointToCollider;
+                                        Collider collider = colliders[colliderIndex];
+                                        // Policy: in order for an collider to be near interactable it must have
+                                        // a NearInteractionGrabbable component on it.
+                                        // FIXME: This is assuming only the grab pointer is using SceneQueryType.SphereOverlap,
```

Comment:
It would be great to file an issue and reference the issue number here, to ensure we do not forget.

Reply:
I believe this one is tracked by https://github.com/microsoft/MixedRealityToolkit-Unity/issues/3758